github/src/main/java/io/syndesis/git/GitWorkflow.java

```diff
@@ -0,0 +1,131 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.syndesis.core.SyndesisServerException;
+
+public class GitWorkflow {
+
+    private static final Logger log = LoggerFactory.getLogger(GitWorkflow.class);
+    /**
+     * Commits and pushes the files to the Git Repository
+     * 
+     * @param repo- an existing github repository
+     * @param message- commit message
+     * @param files- map of file paths along with their content
+     * @param credentials- Git credentials, for example username/password, authToken, personal access token
+     * @return RevCommit, the commit info
+     */
+    public RevCommit createFiles(String repoHtmlUrl, String repoName, String message, Map<String, byte[]> files,
+            UsernamePasswordCredentialsProvider credentials) {
+
+        try {
+            // create temporary directory
+            Path workingDir = Files.createTempDirectory(repoName);
+            if (log.isDebugEnabled()) log.debug("Created temporary directory {}", workingDir.toString());
+
+            // git init
+            Git git = Git.init().setDirectory(workingDir.toFile()).call();
+            writeFiles(workingDir, files);
+
+            RemoteAddCommand remoteAddCommand = git.remoteAdd();
+            remoteAddCommand.setName("origin");
+            remoteAddCommand.setUri(new URIish(repoHtmlUrl));
+            remoteAddCommand.call();
+
+            RevCommit commit = commitAndPush(git, message, credentials);
+            workingDir.toFile().delete();
+            return commit;
+
+        } catch (Exception e) {
+            throw SyndesisServerException.launderThrowable(e);
+        }
+    }
+    
+    public RevCommit updateFiles(String repoHtmlUrl, String repoName, String message, Map<String, byte[]> files,
+            UsernamePasswordCredentialsProvider credentials) {
+
+        try {
+            // create temporary directory
+            Path workingDir = Files.createTempDirectory(repoName);
+            if (log.isDebugEnabled()) log.debug("Created temporary directory {}", workingDir.toString());
+
+            // git clone
+            Git git = Git.cloneRepository().setDirectory(workingDir.toFile()).setURI(repoHtmlUrl).call();
+            writeFiles(workingDir, files);
+
+            RevCommit commit =  commitAndPush(git, message, credentials);
+            
+            // cleanup tmp dir
+            workingDir.toFile().delete();
+            
+            return commit;
+        } catch (Exception e) {
+            throw SyndesisServerException.launderThrowable(e);
+        }
+    }
+    /**
+     * Write files to the file system
+     * 
+     * @param workingDir
+     * @param files
+     * @throws IOException
+     */
+    private void writeFiles(Path workingDir, Map<String, byte[]> files) throws IOException {
+        for (String filePath : files.keySet()) {
+            File file = new File(workingDir.toString() + File.separator + filePath);
+            file.getParentFile().mkdirs();
```

Comment:
Shouldn't the return value be checked here and an exception thrown if its false ?